sally/models/weather.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+
+from datetime import datetime, time
+import requests
+from odoo.exceptions import ValidationError
+from odoo import models, fields, api
+
+
+class StockWarehouseWeather(models.Model):
+    _name = 'stock.warehouse.weather'
+    _description = 'Stock Warehouse Weather'
+    _order = "capture_time"
+
+    warehouse_id = fields.Many2one("stock.warehouse", string="Warehouse")
+
+    temp = fields.Float(string="Temperature")
+    pressure = fields.Float(string="Pressure")
+    humidity = fields.Float(string="Humidity")
+    wind_speed = fields.Float(string="Wind speed")
+    rain_volume = fields.Float(string="Rain volume (for the past hour in mm)")
+    description = fields.Char(string="Description")
+    capture_time = fields.Datetime(string="Capture Time")
+
+
+class StockWarehouse(models.Model):
+    _inherit = "stock.warehouse"
+
+    weather_ids = fields.One2many("stock.warehouse.weather", "warehouse_id")
+
+    def _get_api_key_and_location(self, show_error):
+        api_key = self.env["ir.config_parameter"].sudo(
+        ).get_param("flower_shop.weather_api_key")
+        if api_key == "unset" or not api_key:
+            raise ValidationError(
+                "API key is invalid. Please make sure you have set system parameters correctly.")
+        if not self.partner_id or not self.partner_id.partner_latitude or not self.partner_id.partner_longitude:
+            raise ValidationError(f"Warehouse location is not set properly for {self.name}.")
+        return api_key, self.partner_id.partner_latitude, self.partner_id.partner_longitude
```

Comment:
works perfectly fine. 

same as my previous comment this can be done in a slightly better way, you can check the solution for it